### PR TITLE
fix: support back pressed operation for default permission UI 

### PR DESCRIPTION
**Pull request by khairul-alam-licon**
# Description
Previously, there was no support for back pressed when opening default permission UI. It should be the same operation while users press "Close" button. This PR added the fix for back button pressed.

## Links
- MINI-2033

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [ ] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/permission/ui/MiniAppCustomPermissionWindow.kt [hunk 1]
```diff
@@ -107,8 +108,19 @@ internal class MiniAppCustomPermissionWindow(
 
         customPermissionLayout.findViewById<TextView>(R.id.permissionCloseWindow)
             .setOnClickListener {
-                customPermissionBridgeDispatcher.sendCachedCustomPermissions()
-                customPermissionAlertDialog.dismiss()
+                onNoPermissionsSaved()
             }
+
+        customPermissionAlertDialog.setOnKeyListener { _, keyCode, event ->
+            if (keyCode == KeyEvent.KEYCODE_BACK && event.action == KeyEvent.ACTION_DOWN) {
```
Comment (corycaywood) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/permission/ui/MiniAppCustomPermissionWindow.kt:115]: It seems a little strange to trigger this on `ACTION_DOWN`. Usually I expect a button to trigger only after I've released it, so on `ACTION_UP`.  What do you think @minh-rakuten @khairul-alam-licon ?

Comment (khairul-alam-licon) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/permission/ui/MiniAppCustomPermissionWindow.kt]: Yes, `KeyEvent.ACTION_UP` makes more sense in this usecase. I have updated this part.

**Review by corycaywood — APPROVED** [review]
Looks good 👍

**Review by minh-rakuten — APPROVED** [review]
While I'm playing with permission dialog, I have this crash:
```
com.rakuten.tech.mobile.miniapp.testapp.debug E/Parcel: Class not found when unmarshalling: androidx.fragment.app.FragmentManagerState
    java.lang.ClassNotFoundException: androidx.fragment.app.FragmentManagerState
```

@khairul-alam-licon @corycaywood The problem is I cannot reproduce so we should notice this.

**Comment by khairul-alam-licon** [comment]
> com.rakuten.tech.mobile.miniapp.testapp.debug E/Parcel: Class not found when unmarshalling: androidx.fragment.app.FragmentManagerState
    java.lang.ClassNotFoundException: androidx.fragment.app.FragmentManagerState

This is an issue I have encountered in other branches as well. It's not related to this PR's change, but we should address it in another PR.

**Pull request merged**